src/v/redpanda/application.cc

```diff
@@ -1466,6 +1466,10 @@ void application::wire_up_and_start(::stop_signal& app_signal, bool test_mode) {
             *controller, group_router));
     }
 
+    if (cd.is_cluster_founder().get()) {
```

Comment:
Does is_cluster_founder go to false after the cluster forms?  Otherwise on a 3 node cluster all the nodes will always hit this early-ready path (not necessarily a problem, just means we don't benefit from the fix in this PR in that scenario)

Reply:
yes, it does go to false right after the bootstrap, there is a check for empty controller directory.